### DISCUSSION TITLE
Missing deployment ID / version skew header for some assets

### DISCUSSION THREAD

**Discussion by shaunwarman** [vercel/next.js#73110] [App Router]
It looks as though, even using the [deploymentId configuration](https://nextjs.org/docs/app/building-your-application/deploying#version-skew) option for [version skew support](https://nextjs.org/docs/app/building-your-application/deploying#version-skew), some assets do not have a `?dpl` query param or a `X-Deployment-ID` header included in the request / URI. 

How does version skew protection work for these assets?

<img width="717" alt="Screenshot 2024-11-22 at 3 52 58 PM" src="https://github.com/user-attachments/assets/4a3ea859-1090-4e0a-812a-04dcfb6e4bb7">

**Comment by shaunwarman** [comment]
Actually it looks like next config is used at runtime so we needed to account for that

**Reply by moltar** [reply]
Yeah, this is what I just ran into as well. This is very undocumented. It's not clear at all that the value from the next config also needs to be set at runtime. In my mind, based on other tools, the `*.config.{js,ts}` pattern is always reserved for build-time only. While Next.js seems to try to interpret this config at runtime and pick some values from it. Worst part, all of these values are already available in the build artifact (`required-server-files.json`) that it could get from.

**Reply by coltenkrauter** [reply]
~~Can you please explain further how this works? I added the `deploymentId` to my next.config.mjs and I don't see the header or the query param added.~~

I found my problem... I was using an env var for the deploymentId and it wasn't available at compile time on my GitHub action. All fixed up.
